src/App.tsx

```diff
@@ -1,24 +1,210 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useState, useEffect } from 'react';
+import classNames from 'classnames';
+import { Todo } from './types/Todo';
+import {
+  deleteTodo,
+  getTodos,
+  patchTodo,
+  postTodo,
+} from './api/todos';
+import { TodoList } from './components/TodoList';
 
-const USER_ID = 0;
+const USER_ID = 6616;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState('');
+  const [query, setQuery] = useState('');
+  const [disabledInput, setDisabledInput] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo>();
+  const [loadingIds, setLoadingIds] = useState<number[]>([]);
+
+  const closeError = () => {
+    setError('');
+  };
+
+  const addTodo = (title: string) => {
+    setDisabledInput(true);
+
+    const newTodo = {
+      title,
+      userId: USER_ID,
+      completed: false,
+    };
+
+    setTempTodo({ ...newTodo, id: 0 });
+
+    postTodo(newTodo)
+      .then(result => {
+        setTodos(state => [...state, result]);
+      })
+      .catch(() => {
+        setError('Unable to add a todo');
+        setTimeout(() => {
+          setError('');
+        }, 3000);
+      })
+      .finally(() => {
+        setDisabledInput(false);
+        setTempTodo(undefined);
+      });
+  };
+
+  const removeTodo = (id: number) => {
+    setLoadingIds(state => [...state, id]);
+
+    return deleteTodo(id)
+      .then(() => {
+        setTodos(todos.filter(todo => todo.id !== id));
+      })
+      .catch(() => {
+        setError('Unable to delete a todo');
+        setTimeout(() => {
+          setError('');
+        }, 3000);
+      })
+      .finally(() => {
+        setLoadingIds([]);
```

Comment:
why do you clear all ids here? let's filter them like you did in `update` and create a method for it